src/Form/DataField/DataLinkFieldType.php

```diff
@@ -299,9 +299,7 @@ public function modelTransform($data, FieldType $fieldType): DataField
             }
 
             if (\is_array($data)) {
-                if (0 == \count($data)) {
```

Comment:
Why remove the cleaning of empty array?

Reply:
At this point the $data array can not be null, because the code before is allready checking for empty($data);

```php
if (\is_string($data)) {
                return $out;
            }
            if (empty($data)) {
                $out->setRawData(null);

                return $out;
            }

            if (\is_array($data)) {
                if (\is_string($data[0])) {
                    $out->setRawData($data[0]);
                    if (\count($data) > 1) {
                        $out->addMessage('Data converted into string, some data might have been lost');
                    }
                }
            } else {
                $out->setRawData(null);
                $out->addMessage('Data was not able to be imported: '.\json_encode($data));
            }
````